src/Uno.UI/UI/Xaml/Controls/ContentDialog/ContentDialog.Visuals.cs

```diff
@@ -0,0 +1,402 @@
+					{
+						buttonVisibilityState = "PrimaryVisible";
+					}
+					else if (hasSecondary)
+					{
+						buttonVisibilityState = "SecondaryVisible";
+					}
+					else if (hasClose)
+					{
+						buttonVisibilityState = "CloseVisible";
+					}
+
+					GoToState(useTransitions, buttonVisibilityState);
+				}
+
+				// DefaultButtonStates
+				{
+					var defaultButtonState = "NoDefaultButton";
+
+					var defaultButton = DefaultButton;
+
+					if (defaultButton != ContentDialogButton.None)
+					{
+						// UNO TODO
+						//ctl::ComPtr<DependencyObject> focusedElement;
+						//IFC_RETURN(GetFocusedElement(&focusedElement));
+
+						//BOOLEAN isFocusInCommandArea = FALSE;
+						//IFC_RETURN(m_tpCommandSpacePart.Cast<Grid>()->IsAncestorOf(focusedElement.Get(), &isFocusInCommandArea));
+
+						// If focus is not in the command area, set the default button visualization just based on the property value.
+						// If focus is in the command area, set the default button visualization only if it has focus.
+						if (defaultButton == ContentDialogButton.Primary)
+						{
+							//if (!isFocusInCommandArea || ctl::are_equal(m_tpPrimaryButtonPart.Get(), focusedElement.Get()))
+							{
+								defaultButtonState = "PrimaryAsDefaultButton";
+							}
+						}
+						else if (defaultButton == ContentDialogButton.Secondary)
+						{
+							//if (!isFocusInCommandArea || ctl::are_equal(m_tpSecondaryButtonPart.Get(), focusedElement.Get()))
+							{
+								defaultButtonState = "SecondaryAsDefaultButton";
+							}
+						}
+						else if (defaultButton == ContentDialogButton.Close)
+						{
+							//if (!isFocusInCommandArea || ctl::are_equal(m_tpCloseButtonPart.Get(), focusedElement.Get()))
+							{
+								defaultButtonState = "CloseAsDefaultButton";
+							}
+						}
+					}
+
+					GoToState(useTransitions, defaultButtonState);
+				}
+			}
+
+			{
+				// DialogShowingStates
+				if (m_placementMode == PlacementMode.InPlace)
+				{
+					GoToState(true, m_isShowing && !m_hideInProgress ? "DialogShowing" : "DialogHidden");
+				}
+				else if (m_placementMode != PlacementMode.Undetermined)
+				{
+					// For ContentDialog's shown in the popup, set the state to always showing since the opened
+					// state of the popup effectively controls whether its showing it not.
+					GoToState(false, "DialogShowingWithoutSmokeLayer");
+				}
+
+				// DialogSizingStates
+				{
+					GoToState(useTransitions, fullSizeDesired ? "FullDialogSizing" : "DefaultDialogSizing");
+				}
+
+				// DialogBorderStates
+				{
+					GoToState(useTransitions, "NoBorder");
+				}
+			}
+
+			// On PhoneBlue, the dialog did not move out of the way of the input pane.
+			if (true)
+			{
+				AdjustVisualStateForInputPane();
+			}
+		}
+
+		private void ResetAndPrepareContent()
+		{
+			// Uno TODO: understand when this is applicable (seemingly while dialog is open?)
+			//if (HasValidAppliedTemplate() && m_tpCurrentAsyncOperation && m_hasPreparedContent)
+			//{
```

Comment:
Codacy found an issue: [Remove this commented out code.](https://app.codacy.com/gh/unoplatform/uno/pullRequest?prid=6389897)